spdx-utils/src/main/kotlin/SpdxLicenseException.kt

```diff
@@ -88,6 +90,8 @@ enum class SpdxLicenseException(
         /**
          * Return the enum value for the given [id], or null if it is no SPDX license exception id.
          */
+        @JsonCreator(mode = JsonCreator.Mode.DELEGATING)
```

Comment:
Why is this necessary? I have problems understanding where it is used. Maybe explain it in the commit message.

Reply:
Without this, the license / exception enums would require their *names* to be used instead of their *ids* for deserialization, and the point of this change is to not use ORT-specific enum names in the `yml`-mapping files anymore, so these files could be used by other parties, too.

Reply:
Yes, but my problem is still to find whereis this constructor used for deserialization? I guess the one for `SpdxLicense` is used here:
https://github.com/oss-review-toolkit/ort/blob/d4f0cb598cc9af4402be6c4f5ee599a6c8136bea/spdx-utils/src/main/kotlin/SpdxSimpleLicenseMapping.kt#L34-L37

But where is the one for exceptions used?

Reply:
> I guess the one for `SpdxLicense` is used here:

Correct.

> But where is the one for exceptions used?

Nowhere. But since both files are generated from the same template by a Gradle task, they both contain these lines. But having them also in `SpdxLicenseException.kt` shouldn't hurt, and is more consistent.